### PR TITLE
Fix #4011: Handle more huggable immediately nested parens/brackets.

### LINKED ISSUE #4011: Preview: handle more huggable immediately nested parens/brackets
**Describe the style change**

#3964 handled the most common case from #1811, and I'd like the following cases to be handled too:

```python
foo(
    (
        1,
        2,
        3,
    )
)

foo([
    [
        1,
        2,
        3,
    ]
])
```

See [playgroud](https://black.vercel.app/?version=main&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4ADeAGddAD2IimZxl1N_WlkOE2WKNpiCxsiZ6ZIRL1cm3cr3OmQofsf9OUyzIBO_A83ZWkx59NLlC9fXK-w-pVBSBaFpfY2W8u9h4XGb9FVyhxmqoQVyfT5MsLdoUaBIqIvVCpha-di3Y3IZYwAAAJljAbaAgB-SAAGDAd8BAADjWOQZscRn-wIAAAAABFla).

**Desired style**

```python
foo((
    1,
    2,
    3,
))

foo([[
    1,
    2,
    3,
]])
```

**Additional context**

I'll send a PR if this proposal looks good.

### PR DESCRIPTION
**Pull request by yilei**
### Description

Fix #4011.

### Checklist - did you ...

- [x] Add an entry in `CHANGES.md` if necessary?
- [x] Add / update tests if necessary?
- [x] Add new / update outdated documentation?

### CODE REVIEW

**Comment by yilei** [comment]
Looks like the changelog checker doesn't like me combining two PRs in CHANGES.md, let me know if you prefer separating them.

File: CHANGES.md [hunk 1]
```diff
@@ -13,8 +13,8 @@
 
 ### Preview style
 
-- Multiline dictionaries and lists that are the sole argument to a function are now
-  indented less (#3964)
+- Multiline tuples, lists, and dictionaries that are the sole argument to a function or
+  inside another tuple, list, or dictionary are now indented less (#3964, #4012)
```
Comment (JelleZijlstra) [inline CHANGES.md:17]:
```suggestion
  inside another tuple, list, or dictionary are now indented less (#3964) (#4012)
```

File: src/black/lines.py [hunk 2]
```diff
@@ -200,9 +200,9 @@ def is_triple_quoted_string(self) -> bool:
         value = self.leaves[0].value
         if value.startswith(('"""', "'''")):
             return True
-        if Preview.accept_raw_docstrings in self.mode and value.startswith(
-            ("r'''", 'r"""', "R'''", 'R"""')
```
Comment (JelleZijlstra) [inline src/black/lines.py:204]: hm, here the old formatting seems better

Comment (JelleZijlstra) [reply src/black/lines.py:204]: Perhaps the rule should be that if the inner collection fits on a single line, we keep the parentheses with the inner collection.

Comment (yilei) [reply src/black/lines.py]: Agreed. Updated so it no longer hugs if the pre-hugged version fits on a single line.

**Comment by yilei** [comment]
Looking at diff shades, this explodes:

```diff
-man_pages = [
-    ("index", doc_module.__name__, doc_module.__name__ + " Documentation", [""], 1)
-]
+man_pages = [(
+    "index",
+    doc_module.__name__,
+    doc_module.__name__ + " Documentation",
+    [""],
+    1,
+)]
```

```diff
-    c = Context({
-        "version": get_docs_version(),
-    })
+    c = Context(
+        {
+            "version": get_docs_version(),
+        }
+    )
```

I'll take a look later.

**Comment by yilei** [comment]
diff-shades finally succeeded after several retries.

I tried my best to read through the preview diffs, and they all look expected to me.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**